lib/vfscore/automount.c

```diff
@@ -385,6 +465,26 @@ static int vfscore_extract_volume(const struct vfscore_volume *vv)
 	if (unlikely(vlen == 0))
 		uk_pr_warn("Initrd \"%s\" seems to be empty.\n", vv->sdev);
 
+	/* parse mount options */
+	opts = vfscore_extract_parse_opts(vv->opts);
+
+	/*
+	 * Cases for mount a RamFS volume (see `vfscore_extract_parse_opts()`)
+	 * - ramfs=1: Always
+	 * - ramfs=2: Only if there is nothing else mounted
+	 */
+	if ((opts & LIBVFSCORE_EXTRACTOPT_MOUNT_RAMFS) &&
+	    (!(opts & LIBVFSCORE_EXTRACTOPT_MOUNT_IFNOMOUNT) ||
+	     uk_list_empty_careful(&mount_list))) {
+		uk_pr_info("Mounting 'ramfs' to %s...\n", vv->path);
+		rc = mount("", vv->path, "ramfs", 0x0, "");
+		if (unlikely(rc)) {
+			uk_pr_crit("Failed to mount ramfs to \"%s\": %d\n",
+				   vv->path, errno);
+			return -1;
```

Comment:
Why not just return `rc`?

Reply:
I think this is a detail. `mount` will return -1 in error cases. If I want to forward an meaningful error, I should forward `-errno`. THis is to make clear that I am not blindly taking whatever mount returns.